pages/game.tsx

```diff
@@ -1,22 +1,127 @@
-import React from "react";
-import Link from "next/link";
-import { Container } from "react-bootstrap";
+import Board from "../components/Board";
+import Keyboard from "../components/Keyboard";
+import { createContext, useState, Dispatch, SetStateAction } from "react";
+import { boardDefault, words } from "../components/words";
+import GameOver from "../components/GameOver";
+interface IContext {
+  board: string[][];
+  setBoard: (board: string[][]) => void;
+  currAttempt: {
+    rowIndex: number;
+    letterIndex: number;
+  };
+  setCurrAttempt: (currAttempt: {
+    rowIndex: number;
+    letterIndex: number;
+  }) => void;
+  onEnter: () => void;
+  onDelete: () => void;
+  onLetter: (value: string) => void;
+  correctWord: string;
+  disabledLetters: string[];
+  setDisabledLetters: Dispatch<SetStateAction<string[]>>;
+  correctLetters: string[];
+  setCorrectLetters: Dispatch<SetStateAction<string[]>>;
+  closeLetters: string[];
+  setCloseLetters: Dispatch<SetStateAction<string[]>>;
+  gameOver: { gameOver: boolean; guessedWord: boolean };
+  setGameOver: (gameOver: { gameOver: boolean; guessedWord: boolean }) => void;
+}
+
+export const GameContext = createContext<Partial<IContext>>({});
```

Comment:
Interesting use of React Context here to keep the tree of props cleaner. Consider moving your default values to inside this call so you don't need to use a `Partial` interface wrapper. This would also hopefully eliminate the need for the assert defined symbols (`!`) that you have seemingly everywhere.